concrete-core/src/backends/core/implementation/engines/lwe_ciphertext_vector_inplace_encryption.rs

```diff
@@ -0,0 +1,169 @@
+use concrete_commons::dispersion::Variance;
+
+use crate::backends::core::implementation::engines::CoreEngine;
+use crate::backends::core::implementation::entities::{
+    LweCiphertextVector32, LweCiphertextVector64, LweSecretKey32, LweSecretKey64,
+    PlaintextVector32, PlaintextVector64,
+};
+use crate::specification::engines::{
+    LweCiphertextVectorInplaceEncryptionEngine, LweCiphertextVectorInplaceEncryptionError,
+};
+use crate::specification::entities::{
+    LweCiphertextVectorEntity, LweSecretKeyEntity, PlaintextVectorEntity,
+};
+
+impl
+    LweCiphertextVectorInplaceEncryptionEngine<
+        LweSecretKey32,
+        PlaintextVector32,
+        LweCiphertextVector32,
+    > for CoreEngine
+{
+    /// # Example:
+    /// ```
+    /// use concrete_commons::dispersion::Variance;
+    /// use concrete_commons::parameters::{LweCiphertextCount, LweDimension};
+    /// use concrete_core::prelude::*;
+    /// # use std::error::Error;
+    /// #
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// let mut engine = CoreEngine::new()?;
+    /// // DISCLAIMER: the parameters used here are only for test purpose, and not secure.
+    /// let lwe_dimension = LweDimension(6);
+    /// // Here a hard-set encoding is applied (shift by 20 bits)
+    /// let input = vec![3_u32 << 20; 3];
+    /// let noise = Variance(2_f64.powf(-25.));
+    /// let key_1: LweSecretKey32 = engine.generate_lwe_secret_key(lwe_dimension)?;
+    /// let plaintext_vector: PlaintextVector32 = engine.create_plaintext_vector(&input)?;
+    /// let mut ciphertext_vector: LweCiphertextVector32 =
+    ///     engine.encrypt_lwe_ciphertext_vector(&key_1, &plaintext_vector, noise)?;
```

Comment:
Sorry @tmontaigu I don't understand what you mean by manually? I could create a ciphertext vector encrypting zeros instead?

Reply:
Yes, that's possible. Actually the idea of the `inplace` operations is to really drop what was inside the original ciphertext anyway, so encrypting the same message with another key, or another message with the same key both work.

Reply:
Maybe I should add a comment that states that in the example code, saying that we show the first case but that the second case is also possible.

Reply:
Then I think for inplace operation we can just use the same key to encrypt the same message or create an encryption of zero, it would be less confusing in my opinion to have only one key used than two when having only one is still correct.